Add command line docs (#337)
timescale/promscale: docs/cli.md

# Promscale command line doc

This document gives you information about the configuration flags supported by promscale.
You can also find information on flags with `promscale_<version> -help` 

## General flags
| Flag | Type | Default | Description |
|------|:-----:|:-------:|:-----------:|
| install-timescaledb | boolean | true | Install or update TimescaleDB extension. |

> | leader-election-pg-advisory-lock-id | integer | 0 | Leader-election based high-availability. It is based on PostgreSQL advisory lock and requires an unique advisory lock ID per high-availability group. |

Reviewer: ```suggestion
| leader-election-pg-advisory-lock-id | integer | 0 (disabled) | Leader-election based high-availability. It is based on PostgreSQL advisory lock and requires a unique advisory lock ID per high-availability group. Only a single connector in each high-availability group will write data at one time. A value of 0 disables leader election. |
```

> | leader-election-pg-advisory-lock-prometheus-timeout | slack/duration | -1 | Prometheus timeout duration for leader-election high-availability. The connector will resign if the associated Prometheus instance does not respond within the given timeout. Note: Make sure that only one Prometheus instance talks to the adapter. Timeout value should be co-related with Prometheus scrape interval but enough to prevent random flips. |

Reviewer: ```suggestion
| leader-election-pg-advisory-lock-prometheus-timeout | slack/duration | -1 | Prometheus timeout duration for leader-election high-availability. The connector will resign if the associated Prometheus instance does not respond within the given timeout. This value should be a low multiple of the Prometheus scrape interval, big enough to prevent random flips. |
```

> | leader-election-pg-advisory-lock-prometheus-timeout | slack/duration | -1 | Prometheus timeout duration for leader-election high-availability. The connector will resign if the associated Prometheus instance does not respond within the given timeout. This value should be a low multiple of the Prometheus scrape interval, big enough to prevent random flips. |

Reviewer: ```suggestion
| leader-election-pg-advisory-lock-prometheus-timeout | slack/duration | -1 | Prometheus timeout duration for leader-election high-availability. The connector will resign if the associated Prometheus instance does not send any data within the given timeout. This value should be a low multiple of the Prometheus scrape interval, big enough to prevent random flips. |
```

| log-format | string | logfmt | Log format to use from [ "logfmt", "json" ]. |
| log-level | string | debug | Log level to use [ "error", "warn", "info", "debug" ]. |

> | migrate | string | true | Update the Prometheus SQL to the latest version. Valid options are: [true, false, only]. |

Reviewer: ```suggestion
| migrate | string | true | Update the Prometheus SQL  schema to the latest version. Valid options are: [true, false, only]. |
```

> | read-only | boolean | false | Read-only mode for the connector. Operations related to writing or updating the database are restricted. It is useful when pointing adapter to the read replica. |

Reviewer: ```suggestion
| read-only | boolean | false | Read-only mode for the connector. Operations related to writing or updating the database are disallowed. It is used when pointing the connector to a TimescaleDB read replica. |
```

| migrate | string | true | Update the Prometheus SQL schema to the latest version. Valid options are: [true, false, only]. |
| read-only | boolean | false | Read-only mode for the connector. Operations related to writing or updating the database are disallowed. It is used when pointing the connector to a TimescaleDB read replica. |

> | scheduled-election-interval | duration | 5 seconds | Interval at which scheduled election runs. This is used to select a leader and confirm that we still holding the advisory lock. |

Reviewer: should this be renamed to `leader-election-scheduled-interval`?

Author: Yes, sounds more relateable.

| web-listen-address | string | :9201 | Address to listen on for web endpoints. |
| web-telemetry-path | string | `/metrics` | Web endpoint for exposing Promscale's Prometheus metrics. |

## Database flags

| Flag | Type | Default | Description |
|------|:-----:|:-------:|:-----------:|

> | async-acks | boolean | false | Acknowledge asynchronous inserts. If this is true, the inserter will not wait after insertion of metric data in the database.|

Reviewer: ```suggestion
| async-acks | boolean | false | Acknowledge asynchronous inserts. If this is true, the inserter will not wait after insertion of metric data in the database. This increases throughput at the cost of a small chance of data loss. | 
```

Author: How is there gonna be a data loss? Do we discard samples after a time window?

Reviewer: If the insert is issued, the ack, is returned, and the DB crashes before the insert is written to disk. That's why it's off by default.

| db-connect-retries | integer | 0 | Number of retries Promscale should make for establishing connection with the database. |
| db-connections-max | integer | 80% of possible connections db | Maximum number of connections to the database that should be opened at once. It defaults to 80% of the maximum connections that the database can handle. |
| db-host | string | localhost | Host for TimescaleDB/Vanilla Postgres. |
| db-name | string | timescale | Database name. |
| db-password | string | | Password for connecting to TimescaleDB/Vanilla Postgres. |
| db-port | int | 5432 | TimescaleDB/Vanilla Postgres connection password. |
| db-ssl-mode | string | require | TimescaleDB/Vanilla Postgres connection ssl mode. If you do not want to use ssl, pass `allow` as value. |
| db-user | string | postgres | TimescaleDB/Vanilla Postgres user. |
| db-writer-connection-concurrency | int | 4 | Maximum number of database connections for writing per go process. |

> | labels-cache-size | unsigned-integer | 10000 | Maximum number of labels to cache. |

Reviewer: The cache sizes are general not database flags

| metrics-cache-size | unsigned-integer | 10000 | Maximum number of metric names to cache. |

> | tput-report | integer | 0 | Interval in seconds at which throughput should be reported. |

Reviewer: ```suggestion
| tput-report | integer | 0 (disabled) | Interval in seconds at which throughput should be reported. |
```

> | tput-report | integer | 0 (disabled) | Interval in seconds at which throughput should be reported. |

Reviewer: This is a general not database flag. move to section above.

Author: Sure. I wondered to keep this at general, but then the use case felt me like it denoted the ingestion rate (as fae as I recall) of the timescaledb, so I left it in the database flag. Anyways, moving this to general section.
